### PR TITLE
add to cpu vecops the function execute_program

### PR DESCRIPTION
**Pull request by mickeyasa**
add a vector operation to cpu backend that takes a vector of array and execute a program element wise

### CODE REVIEW

File: icicle/backend/cpu/src/field/cpu_vec_ops.cpp [hunk 1]
```diff
@@ -838,6 +841,34 @@ eIcicleError cpu_highest_non_zero_idx(
...
+  for (uint64_t i = 0; i < total_nof_operations; i++) {
+    prog_executor.execute();
+    for (auto& var_ptr : prog_executor.m_variable_ptrs) {
+      var_ptr++;
+    }
+  }
+  return eIcicleError::SUCCESS;
+}
+
+// REGISTER_EXECUTE_PROGRAM_BACKEND("CPU", cpu_execute_program<scalar_t>);
```
Comment (yshekel) [inline icicle/backend/cpu/src/field/cpu_vec_ops.cpp:870]: why not make it a real API ? want  Idan to do it?

Comment (mickeyasa) [reply icicle/backend/cpu/src/field/cpu_vec_ops.cpp:870]: exactly

File: icicle/backend/cpu/src/field/cpu_vec_ops.cpp [hunk 2]
```diff
@@ -10,6 +10,9 @@
 #include <sys/types.h>
 #include <vector>
 
+#include "icicle/program/program.h"
+#include "../../icicle/backend/cpu/include/cpu_program_executor.h"
```
[the remark this answers is not in the document]
Comment (mickeyasa) [reply icicle/backend/cpu/src/field/cpu_vec_ops.cpp:14]: fixed

File: icicle/backend/cpu/src/field/cpu_vec_ops.cpp [hunk 3]
```diff
@@ -329,7 +332,7 @@ class VectorOpTask : public TaskBase
 
   // An array of available function pointers arranged according to the VecOperation enum
   using FunctionPtr = void (VectorOpTask::*)();
-  static constexpr std::array<FunctionPtr, static_cast<int>(NOF_OPERATIONS)> functionPtrs = {
+  static constexpr std::array<FunctionPtr, static_cast<int>(NOF_VECOPS_OPERATIONS)> functionPtrs = {
```
Comment (danny-shterman) [inline icicle/backend/cpu/src/field/cpu_vec_ops.cpp:335]: VECOPS_OPERATIONS = VEC OPERATIONS OPERATION
Why not NOF_VEC_OPERATIONS?

**Review by danny-shterman — APPROVED** [review]

File: icicle/backend/cpu/src/field/cpu_vec_ops.cpp (hunk 3 above)
Comment (mickeyasa) [reply icicle/backend/cpu/src/field/cpu_vec_ops.cpp:335]: Fixed

**Review by yshekel — APPROVED** [review]

**Pull request merged**